integration-tests/downscope/downscope_test.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2021 Google LLC.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package downscope
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"google.golang.org/api/option"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/google/downscope"
+	storage "google.golang.org/api/storage/v1"
+)
+
+const (
+	envServiceAccountFile = "GCLOUD_TESTS_GOLANG_KEY"
+	rootTokenScope        = "https://www.googleapis.com/auth/cloud-platform"
+)
+
+var (
+	object1 = "cab-first-c45wknuy.txt"
+	object2 = "cab-second-c45wknuy.txt"
+	bucket1 = "dulcet-port-762"
+
+	rootCredential *google.Credentials
+)
+
+// TestMain contains all of the setup code that needs to be run once before any of the tests are run
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if testing.Short() {
+		// This line runs all of our individual tests
+		os.Exit(m.Run())
+	}
+	ctx := context.Background()
+	credentialFileName := os.Getenv(envServiceAccountFile)
+	credentialsFileData, err := os.ReadFile(credentialFileName)
+	if err != nil {
+		log.Fatalf("failed to open credentials file: %v", err)
+	}
+	rootCredential, err = google.CredentialsFromJSON(ctx, credentialsFileData, rootTokenScope)
```

Comment:
You could use https://pkg.go.dev/google.golang.org/api/transport#Creds with the scope and creds file option to simplify this a bit.